cmd/nerdctl/container_run_mount.go

```diff
@@ -189,6 +191,25 @@ func generateMountOpts(ctx context.Context, cmd *cobra.Command, client *containe
 					return nil, nil, nil, err
 				}
 			}
+		} else if runtime.GOOS == "linux" {
+			for _, m := range mounts {
+				m := m
+				defer unmounter(m.Source)
+				if m.Type == "bind" && userns.RunningInUserNS() {
+					// For https://github.com/containerd/nerdctl/issues/2056
+					unpriv, err := mountutil.UnprivilegedMountFlags(m.Source)
```

Comment:
Should we have an integ or an unit test for this that the unpriv mount flags are carried over?
If my understanding is correct; a user will still have to ensure that the right subuid and subgid ranges are specified.

Reply:
Ideally yes, but not sure how to do that